src/visualization/waterSurface/WaterSurfaceMesh.h

```diff
@@ -53,7 +53,7 @@ class WaterSurfaceMesh : public SceneBase3D::Object3D, public SceneGraph::Drawab
   // std::vector<VertexData> newData = _data;
 
 #pragma omp parallel for
-    for (size_t i = 0; i < _data.size(); i++) {
+    for (int i = 0; i < _data.size(); i++) {
```

Comment:
Why? An int is not safe when iterating on a STL container, `size_t` is designed for this.
Ideally you would even use `std::vector<Foo>::size_type` (see https://stackoverflow.com/questions/45168636/stdsize-t-or-stdvectorfoosize-type) but most of the time `std::size_t` is fine (certainly better than a int).

Reply:
I found that MSVC even 2019 doesn't support OpenMP 3.0 so all for loops need to use a signed type. I disabled OpenMP on MSVC below in 49b116f, performance seemed very similar.